packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -374,7 +374,8 @@ function patchShadowRoot(vm: VM, newCh: VNodes) {
             );
         }
     }
-    if (vm.state === VMState.connected) {
+
+    if (vm.state === VMState.connected && isFalse(renderer.ssr)) {
```

Comment:
The body of `runRenderedCallback()` seems like a better place for this for this guard.

```
function runRenderedCallback(vm: VM) {
    if (isTrue(vm.renderer.ssr)) {
        return;
    }
    const { rendered } = Services;
    if (rendered) {
        invokeServiceHook(vm, rendered);
    }
    invokeComponentRenderedCallback(vm);
}
```